README.md

```diff
@@ -288,18 +289,71 @@ public class FailoverSampleApp2 {
   }
 }
 ```
+
 >### :warning: Warnings About Proper Usage of the AWS JDBC Driver for MySQL
 >1. A common practice when using JDBC drivers is to wrap invocations against a Connection object in a try-catch block, and dispose of the Connection object if an Exception was hit. If this practice is left unaltered, the application will lose the fast-failover functionality offered by the Driver. When failover occurs, the Driver internally establishes a ready-to-use connection inside the original Connection object before throwing an exception to the user. If this Connection object is disposed of, the newly established connection will be thrown away. The correct practice is to check the SQL error code of the exception and reuse the Connection object if the error code indicates successful failover. [FailoverSampleApp1](#sample-code) and [FailoverSampleApp2](#sample-code-1) demonstrate this practice. See the section below on [Failover Exception Codes](#failover-exception-codes) for more details.
 >2. It is highly recommended that you use the cluster and read-only cluster endpoints instead of the direct instance endpoints of your Aurora cluster, unless you are confident about your application's usage of instance endpoints. Although the Driver will correctly failover to the new writer instance when using instance endpoints, usage of these endpoints are discouraged because individual instances can spontaneously change reader/writer status when failover occurs. The driver will always connect directly to the instance specified if an instance endpoint is provided, so a write-safe connection cannot be assumed if the application uses instance endpoints.
+
+### Connection Plugin Manager
```

Comment:
I think we may need to rework this section a little bit. It needs a more structured introduction. Most customers looking into this probably don't know what a Connection Plugin Manager is. Consider answering a few of these questions as an outline:

- What's a connection plugin manager? Try to say more than just saying it's a manager for connection plugins. Without context, a question that may come to mind is how is failover associated with a connection plugin?
- What can I do with a connection plugin?
- Why would I need a connection plugin?

If you can give an overview of what a connection plugin manager is and tell me the value it brings, I would be more inclined to learn more about it.

Less is more and I would lead with the diagram and let that do the heavy lifting after the introduction. You can then walk through the diagram in the next paragraph for clarity. Going straight to connections are widgets and how they are chained might be too much, too soon. Picture first and then the explanation. We can probably combine some of this explanations with your explanation below the picture.